packages/cadl-ranch-specs/http/payload/xml/mockapi.ts

```diff
@@ -134,12 +134,6 @@ function createServerTests(uri: string, data?: any) {
         status: 200,
         body: xml(data),
       },
-      handler: (req: MockRequest) => {
```

Comment:
for xml, current request comparison is format sensitive, which cause error:
```
  '<ModelWithEncodedNamesSrc>\n' +
  '  <SimpleModelData>\n' +
  '    <name>foo</name>\n' +
  '    <age>123</age>\n' +
  '  </SimpleModelData>\n' +
  '  <PossibleColors>\n' +
  '    <string>red</string>\n' +
  '    <string>green</string>\n' +
  '    <string>blue</string>\n' +
  '  </PossibleColors>\n' +
  '</ModelWithEncodedNamesSrc>\n'
 Actual:
"<?xml version='1.0' encoding='utf-8'?>\n" +
  '<ModelWithEncodedNamesSrc><SimpleModelData><name>foo</name><age>123</age></SimpleModelData><PossibleColors><string>red</string><string>green</string><string>blue</string></PossibleColors></ModelWithEncodedNamesSrc>'
```

Reply:
@tadelesh This handler removal will cause issue. But, I do not think it is because of the case sensitivity. The previous handler was using `data` for comparison while the generated handler uses the `xml(data)` for comparison which causes the issue. I have added a fix for that and confirmed this is working fine now.